File: python-algorithm/leetcode/problem_661.py
"""661. Image Smoother
https://leetcode.com/problems/image-smoother/
"""
from typing import List


class Solution:
    def imageSmoother(self, img: List[List[int]]) -> List[List[int]]:
        m, n = len(img), len(img[0])
        ans = [[0] * n for _ in range(m)]
        for i in range(m):
            for j in range(n):
                cur, cnt = 0, 0
                for x in range(i - 1, i + 2):
                    for y in range(j - 1, j + 2):
                        if 0 <= x < m and 0 <= y < n:
                            cur += img[x][y]
                            cnt += 1
                ans[i][j] = int(cur / cnt)
        return ans

File: java-algo/src/main/java/com/leetcode/Problem661.java
package com.leetcode;

public class Problem661 {
    public int[][] imageSmoother(int[][] img) {
        int m = img.length, n = img[0].length;
        int[][] ans = new int[m][n];
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                ans[i][j] = cal(img, i, j);
            }
        }
        return ans;
    }

    private int cal(int[][] grid, int x, int y) {
        int sum = 0, cnt = 0;
        int left = Math.max(0, y - 1);
        int right = Math.min(grid[0].length - 1, y + 1);
        int up = Math.max(0, x - 1);
        int down = Math.min(grid.length - 1, x + 1);
        for (int i = up; i <= down ; i++) {
            for (int j = left; j <= right; j++) {
                cnt++;
                sum += grid[i][j];
            }
        }
        return sum / cnt;
    }
}
